contracts/pool/base/PoolBase.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+
+import {IPoolFactory} from "../../interfaces/IPoolFactory.sol";
+import {PoolErrors} from "./PoolErrors.sol";
+
+/// @title Base Pool contract
+/// @author Tigran Arakelyan
+/// @dev Should be inherited
+abstract contract PoolBase is ERC20, ReentrancyGuard, PoolErrors {
+    using SafeERC20 for IERC20;
+
+    IERC20 public immutable asset; // The asset deposited by Lenders into the Pool
+    IPoolFactory public immutable poolFactory; // The Pool factory that deployed this Pool
+
+    struct PoolInfo {
+        uint256 lockupPeriod;
+        uint256 minInvestmentAmount;
+        uint256 investmentPoolSize;
+    }
+
+    PoolInfo public poolInfo;
+    uint256 public totalInvested;
```

Comment:
I never leave such an important vars public - the best approach for me to restrict it to be private, setter internal and getter public.